PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,204 @@
+            }
+
+            public void Reset()
+            {
+                this.pizza = new Pizza();
+            }
+
+            public Pizza getPizza()
+            {
+                Pizza cookedPizza = this.pizza;
+                this.Reset();
+                return cookedPizza;
+            }
+
+        }
+
+
+        public class Pizza
+        {
+            private List<object> pizza_ingredients = new List<object>();
+
+            public void addIngredient(string ingredient)
+            {
+                this.pizza_ingredients.Add(ingredient);
+            }
+
+            public string listIngredients()
+            {
+                string str = "";
+
+                for (int i = 0; i < this.pizza_ingredients.Count; i++)
+                {
+                    str += this.pizza_ingredients[i] + ", ";
+                }
+
+                return "Pizza is ready! it has: \t" + str;
+            }
+
+        }
+
+
+        public class Director
+        {
+            private IPizzaBuilder pizzaBuilder;
+            public IPizzaBuilder Builder
+            {
+                set { this.pizzaBuilder = value; }
+            }
+            public void MakePizzaWithSauce()
+            {
+                this.pizzaBuilder.BuildDough();
+                this.pizzaBuilder.BuildSauce();
+            }
+            public void MakePizzaWithTopping()
+            {
+                this.pizzaBuilder.BuildDough();
+                this.pizzaBuilder.BuildTopping();
+            }
+            public void MakePizzaFull()
+            {
+                this.pizzaBuilder.BuildDough();
+                this.pizzaBuilder.BuildSauce();
+                this.pizzaBuilder.BuildTopping();
+            }
+            public void MakePizzaCustom(string s, string t)
+            {
+                this.pizzaBuilder.BuildDough();
+                this.pizzaBuilder.BuildSauce(s);
+                this.pizzaBuilder.BuildTopping(t);
+            }
+
+
+
+
+        }
+
+
+
+        static void Main(string[] args)
+        {
+
+
+
+
+            var director = new Director();
+            var whitePizza = new WhitePizzaBuilder();
+            var wheatPizza = new WheatPizzaBuilder();
+
+            director.Builder = whitePizza;
+            director.MakePizzaWithSauce();
+            Console.WriteLine(whitePizza.getPizza().listIngredients());
+
+            Console.WriteLine("=======================");
+
+            director.Builder = wheatPizza;
+            director.MakePizzaFull();
+            Console.WriteLine(wheatPizza.getPizza().listIngredients());
+
+            Console.WriteLine("=======================");
+
+            //BONUS 1
+            //pizza without dough....
+            wheatPizza.BuildSauce();
+            wheatPizza.BuildTopping();
+            Console.WriteLine(wheatPizza.getPizza().listIngredients());
+
+            Console.WriteLine("=======================");
+
+            //BONUS 2
```

Comment:
Great Job doing the second Bonus and it was great that you used a user friendly language to interact with the user 👏👏, but it wasn't working right if I entered a white dough there isn't anything printed to the user